### PR TITLE
INTER-2364: Increase code coverage

### PR DESCRIPTION
**Pull request by erayaydin**
### New test files

- `EnumCoverageTest` - parameterized tests covering enum types with `from()`, `tryFrom()`, and `ObjectSerializer::deserialize()`.
  - ⚠️ This is not affecting the code coverage percentage. Because enums do not count.
- `EventAllPropertiesTest` - covers all Event getters, setters, and sub-object property setters
- `LabelsInnerTest` - coverage including range validation, ArrayAccess, serialization

### Expands

- `RawDeviceAttributesTest` - rewritten with all setters, validation boundaries (device_memory, battery_level, screen_resolution), ArrayAccess
- `ObjectSerializerTest` - added edge cases for `sanitizeTimestamp`, `toPathValue`, `toHeaderValue`, `toString`, `serializeCollection`, and `deserialize` branches
- ❗ `FingerprintApiTest` - added `ConnectException` tests, `createHttpClientOption`, `searchEventsRequest` parameter validation, invalid JSON response handling, non-2xx status code paths
- Added `isNullableSetToNull` paths, uncovered setter tests, and validation branches (min, max, etc.) for all Model files

### Removed

Empty or unnecessary enum test files.

### Infrastructure

- Exclude `ModelInterface.php` from coverage (its interface)
- Skip files with 0 coverable statements from coverage report markdown
- Add `@codeCoverageIgnoreStart` (and `End`) with `generate.sh` post-processing for unreachable discriminator null-check in `EventRuleAction`
  - 💬 We can't overwrite the template file because it's also in use for regular (non-discriminator) models and properties.
- Update `.openapi-generator-ignore` with correct entries

### What's Left 🤔 
- `ObjectSerializer.php` - edge cases: discriminator logic, `SplFileObject`/`byte`/`void` types, `deepObject` style
- ❗❕ `FingerprintApi.php` - async error handler callbacks (**we have a bug ⚠️ ⚠️ ⚠️** about type-hinting; fix PR will be created (it expects `ApiException` but receives `ClientException`))
- `Sealed module` - will be covered after PR #254 is merged

### CODE REVIEW

File: test/Model/EnumCoverageTest.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    /**
+     * @param class-string<\BackedEnum> $enumClass
+     */
+    #[DataProvider('enumClassProvider')]
+    public function testTryFromReturnsNullForUnknownValue(string $enumClass): void
+    {
+        $result = $enumClass::tryFrom('__nonexistent_value__');
+        $this->assertNull($result);
+    }
```
[the remark this answers is not in the document]
Comment (erayaydin) [reply test/Model/EnumCoverageTest.php:79]: It's an edge case... unlikely

File: test/Model/EnumCoverageTest.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        foreach ($cases as $case) {
+            $resolved = $enumClass::from($case->value);
+            $this->assertSame($case, $resolved);
+        }
+    }
+
+    /**
+     * @param class-string<\BackedEnum> $enumClass
+     */
+    #[DataProvider('enumClassProvider')]
```
[the remark this answers is not in the document]
Comment (erayaydin) [reply test/Model/EnumCoverageTest.php:74]: Fixed with the commit [#e91b4f8](https://github.com/fingerprintjs/php-sdk/pull/258/commits/e91b4f8270ba8f9ce9623acf30f1b1dc9f1bfa36).

File: scripts/generate.sh [hunk 3]
```diff
@@ -86,3 +86,15 @@ sed_in_place 's/@param  int|null \$limit/@param  int \$limit/' "$API_FILE"
...
+# The `type` property is always set by the constructor, so the null check is dead code.
+EVENT_RULE_ACTION_FILE="src/Model/EventRuleAction.php"
+TYPE_NULL_LINE=$(grep -n "null === \$this->container\['type'\]" "$EVENT_RULE_ACTION_FILE" | cut -d: -f1)
+if [ -n "$TYPE_NULL_LINE" ]; then
+  sed_in_place "${TYPE_NULL_LINE}i\\
+        // @codeCoverageIgnoreStart" "$EVENT_RULE_ACTION_FILE"
+  INVALIDPROPS_LINE=$((TYPE_NULL_LINE + 2))
+  sed_in_place "${INVALIDPROPS_LINE}a\\
+            // @codeCoverageIgnoreEnd" "$EVENT_RULE_ACTION_FILE"
+fi
```
[the remark this answers is not in the document]
Comment (erayaydin) [reply scripts/generate.sh:100]: `generate.sh` starts with `rm -Rf ./src/**` which deletes all source files, then regenerates them fresh. So no chance to duplicate. But I'll add `-m1`.

[the remark this answers is not in the document]
Comment (erayaydin) [reply scripts/generate.sh:100]: Fixed with the commit [#95931d2](https://github.com/fingerprintjs/php-sdk/pull/258/commits/95931d2d95ad9c66df597e1c8251c7c28a6bcfa2).

**Review by Orkuncakilkaya — APPROVED** [review]
